rfcs/20230418-add-pow-graph-op/README.md

```diff
@@ -0,0 +1,256 @@
+| input     | `exponent`       | optional                   | f32,f16,bf16  |
+| attribute | `auto_broadcast` | optional, default: 'numpy' | string        |
+| output    | `dst`            | Required                   | f32,f16,bf16* |
+
+*: input and output must have the same data type.
+
+Pros.
+
+1. compatible with mainstream frameworks.
+2. can add `Pow` only for covering the fuctionality of `Pow` and `Rsqrt`
+
+Cons.
+
+1. framework bridge needs to make -1/2 as the second input tensor for `Rsqrt`.
+2. lose the constant information and optimization for sqrt opeartion during
+   transforming.
+
+### Option 4: Add Both Binary Element-wise Pow and Rsqrt
+
+This option will add two operation in Graph API.
+`Pow` takes two input tensors `src` and `exponent`.
+`Rsqrt` is added as an eltwise operation.
+
+| Pow       | Argument Name    | Required or Optional       | Data Type     |
+|-----------|------------------|----------------------------|---------------|
+| input     | `src`            | Required                   | f32,f16,bf16* |
+| input     | `exponent`       | optional                   | f32,f16,bf16  |
+| attribute | `auto_broadcast` | optional, default: 'numpy' | string        |
+| output    | `dst`            | Required                   | f32,f16,bf16* |
+
+*: input and output must have the same data type.
+
+| Rsqrt  | Argument Name | Required or Optional | Data Type     |
+|--------|---------------|----------------------|---------------|
+| input  | `src`         | Required             | f32,f16,bf16* |
+| output | `dst`         | Required             | f32,f16,bf16* |
+
+*: input and output must have the same data type.
+
+Pros.
+
+1. clearly mapping semantics of `Pow` and `Rsqrt` in Pytorch.
+
+Cons.
+
+1. need to add two new operations: `Pow` and `Rsqrt`
+2. `Pow` cannot be optimized with oneDNN primitive.
+
+### Option 5: Add Binary Element-wise Pow and Use Reciprocal and Sqrt as Rsqrt
+
+The option will add only one operation `Pow`, and use `Reciprocal` and `Sqrt` to
+cover the funcionality of `Rsqrt`.
+`Pow` takes two input tensors `src` and `exponent`.
+`Rsqrt` reuse `Reciprocal` and `Sqrt`: Rsqrt(src) = Reciprocal(Sqrt(src)).
+`Reciprocal` and `Sqrt` are already supported in oneDNN Graph API.
+
+| Pow       | Argument Name    | Required or Optional       | Data Type     |
+|-----------|------------------|----------------------------|---------------|
+| input     | `src`            | Required                   | f32,f16,bf16* |
+| input     | `exponent`       | optional                   | f32,f16,bf16  |
+| attribute | `auto_broadcast` | optional, default: 'numpy' | string        |
+| output    | `dst`            | Required                   | f32,f16,bf16* |
+
+Pros.
+
+1. `Pow` is compatible with mainstream frameworks.
+2. `Rsqrt` reuse current op set. Will keep the op set small.
+
+Cons.
+
+1. framework bridge will need to split `Rsqrt` to two oneDNN Graph operations.
+2. `Pow` cannot be optimized with oneDNN primitive.
```

Comment:
why `Pow` cannot be optimized with oneDNN here

Reply:
in this option, the second input is a runtime tensor. but  oneDNN only support exponent as single constant attribute
https://oneapi-src.github.io/oneDNN/dev_guide_eltwise.html

Reply:
Thanks for the explanation.